Windows Support

I can't build this on Windows because glibc (used by `CommandLineTool`) isn't available. Is it feasible to make this support Windows? Does UCRT adequately fill in the shoes of glibc in this case? It seems like all Swift auto-formatting tools are either incompatible with Windows or unmaintained :(

```
#if os(macOS)
    import Darwin.POSIX
#elseif os(Windows)
    import ucrt
#else
    import Glibc
#endif
```

seems to work. I've been using the tool for a few hours without issues. All the tests pass too, though I don't think any of the tests specifically cover the TTY check. I'm not sure how I could manually test that either.

I get a bunch of `LNK4217` warnings (no idea why) and a warning about `isatty` being deprecated:

```
SwiftFormat-master\CommandLineTool\main.swift:61:27: warning: 'isatty' is deprecated: The POSIX name for this item is deprecated. Instead, use the ISO C and C++ conformant name: _isatty. See online help for details.
private let stderrIsTTY = isatty(STDERR_FILENO) != 0
```

CI could be set up to include a Windows runner, but it seems like your CI is in the middle of a migration right now. It might be tricky to do [this](https://swift.org/download/#using-downloads) setup step in CI though, since the instructions state to do that through a specific shell that sets up VS-specific environment variables.

--- a/CommandLineTool/main.swift
+++ b/CommandLineTool/main.swift
@@ -33,6 +33,8 @@ import Foundation
 
 #if os(macOS)
     import Darwin.POSIX
+#elseif os(Windows)
+    import ucrt
 #else
     import Glibc
 #endif

--- a/Sources/CommandLine.swift
+++ b/Sources/CommandLine.swift
@@ -553,7 +553,7 @@ func processArguments(_ args: [String], in directory: String) -> ExitCode {
                         return URL(fileURLWithPath: cachePath)
                     }
                 #endif
-                return URL(fileURLWithPath: "/var/tmp/")
+                return FileManager.default.temporaryDirectory
             }().appendingPathComponent("com.charcoaldesign.swiftformat")
             do {
                 try manager.createDirectory(at: cacheDirectory, withIntermediateDirectories: true, attributes: nil)

